crates/sui-framework/packages/sui-framework/sources/coin.move

```diff
@@ -260,6 +275,40 @@ module sui::coin {
         )
     }
 
+    /// This creates a new currency, via `create_currency`, but with an extra capability that
+    /// allows for specific addresses to have their coins frozen. Those addresses will not
+    /// be able to interact with the coin
+    public fun create_freezable_currency<T: drop>(
+        witness: T,
+        decimals: u8,
+        symbol: vector<u8>,
+        name: vector<u8>,
+        description: vector<u8>,
+        icon_url: Option<Url>,
+        ctx: &mut TxContext
+    ): (TreasuryCap<T>, FreezeCap<T>, CoinMetadata<T>) {
+        let (treasury_cap, metadata) = create_currency(
+            witness,
+            decimals,
+            symbol,
+            name,
+            description,
+            icon_url,
+            ctx
+        );
+        let package = {
+            let type_name = std::type_name::get_with_original_ids<T>();
```

Comment:
How does `get_with_original_ids` work when it's called during package initialization? Wouldn't the package ID still be 0x0?

Reply:
@lxfind and I looked into this offline, the setting of the package ID happens before init is run